R/spawn_success.R

```diff
@@ -5,24 +5,40 @@
 #' @param egg_to_fry_survival The egg to fry survival rate
 #' @param prob_scour The probability of nest scouring
 #' @param spawn_habitat The available spawning habitat in square meters
-#' @param sex_ratio The female to male spawning ratio, default 0.5
-#' @param redd_size The size of redds including defensible space, default value 9.29 square meters
-#' @param fecundity The number of eggs per female, default value 5522
+#' @param sex_ratio The female to male spawning ratio
+#' @param redd_size The size of redds including defensible space
+#' @param fecundity The number of eggs per female
 #' @source IP-117068
 #' @export
 
 spawn_success <- function(escapement, adult_prespawn_survival, egg_to_fry_survival,
                           prob_scour, spawn_habitat,
-                          sex_ratio = 0.5,
-                          redd_size = 9.29,
-                          fecundity = 5522){
+                          sex_ratio = fallRunDSM::params$spawn_success_sex_ratio,
+                          redd_size = fallRunDSM::params$spawn_success_redd_size,
+                          fecundity = fallRunDSM::params$spawn_success_fecundity){
 
   capacity <- spawn_habitat / redd_size
-  spawner_potential <- escapement * adult_prespawn_survival * sex_ratio
 
-  spawners <- ifelse(spawner_potential > capacity, capacity, spawner_potential)
+  spawner_potential <- if(max(escapement) <= 1000000000) {
+    rbinom(31, round(escapement), (adult_prespawn_survival * sex_ratio))
+  } else {
+    round(escapement * adult_prespawn_survival * sex_ratio)
+  }
+
+  spawners <- if(spawner_potential > capacity) {
```

Comment:
this line is producing warnings, we can't vectorize using `if {} else {}` we will need to switch this to an `ifelse` call.  

```{r}
spawners <- ifelse(spawner_potential > capacity, capacity, spawner_potential)
```
or even just `pmax(spawner_potential, capacity)`

Reply:
pretty sure I wrote this :frowning_face: so we can leave it here and I will add the fix when we merge the stochastic version of these functions? what do you think?

Reply:
I just fixed using `pmax(spawner_potential, capacity)`. That way you don't have to think about it again later.